src/util.rs

```diff
@@ -69,19 +70,35 @@ pub extern "C" fn callback(callback_info: v8::FunctionCallbackInfoPtr_Value) {
         let args = (0..length)
             .map(|i| value::Value::from_raw(&isolate, *callback_info.Args.offset(i)))
             .collect();
-        let info = value::FunctionCallbackInfo {
-            isolate: isolate.clone(),
-            length: length,
-            args: args,
-            this: value::Object::from_raw(&isolate, callback_info.This),
-            holder: value::Object::from_raw(&isolate, callback_info.Holder),
-            new_target: value::Value::from_raw(&isolate, callback_info.NewTarget),
-            is_construct_call: 0 != callback_info.IsConstructCall,
-        };
+
 
         let result = panic::catch_unwind(|| {
+            let holder = value::Object::from_raw(&isolate, callback_info.Holder);
+
+            let internal = if holder.internal_field_count() > 0 {
+                if let Some(external) = holder.get_internal_field(0).into_external() {
```

Comment:
The internal_field_count check ensures that space is allocated and then because this is an Option<External> if the field is not set the get_internal_field(0) call would return a null pointer which rust should map to None.